### ISSUE TITLE
LOBS with PostgreSQL

### ISSUE THREAD

**Issue by BorisDomajnko** [keeps/db-preservation-toolkit#255]
I created the the table with column type bytea and inserted some characters there. 
Creation of a SIARD-2 package (from PostgreSQL): OK
Restore to (to PostgreSQL): OK
Creation a package (from PostgreSQL) with option: --export-external-lobs: OK
  (and I can see folder out_pg1.siard_lobseg_1 with the contents)
Restore to (to PostgreSQL): error
  (I did not use any special option! Is this supposed to work in this way?)

dbptk-app.log.txt:
```
2016-08-25 12:05:20,702 [main] DEBUG (c.d.Main) #########################################################
2016-08-25 12:05:20,704 [main] DEBUG (c.d.Main) #   START-ID-43439e0b-27f5-48ff-9a66-3c49438826f6
2016-08-25 12:05:20,704 [main] DEBUG (c.d.Main) #   START v2.0.0-beta5
2016-08-25 12:05:20,704 [main] DEBUG (c.d.Main) #########################################################
2016-08-25 12:05:20,704 [main] DEBUG (c.d.c.CLI) Operating system: Windows 7
2016-08-25 12:05:20,704 [main] DEBUG (c.d.c.CLI) Architecture: x86
2016-08-25 12:05:20,704 [main] DEBUG (c.d.c.CLI) Version: 6.1
2016-08-25 12:05:20,704 [main] DEBUG (c.d.c.CLI) Java vendor: Oracle Corporation
2016-08-25 12:05:20,704 [main] DEBUG (c.d.c.CLI) Java version: 1.8.0_101
2016-08-25 12:05:20,704 [main] DEBUG (c.d.c.CLI) Java class version: 52.0
2016-08-25 12:05:20,704 [main] DEBUG (c.d.c.CLI) Default Charset reported by java: UTF-8
2016-08-25 12:05:20,704 [main] DEBUG (c.d.c.CLI) Default Charset used by StreamWriter: UTF8
2016-08-25 12:05:20,705 [main] DEBUG (c.d.c.CLI) file.encoding property: UTF-8
2016-08-25 12:05:20,717 [main] DEBUG (c.d.m.Reporter) moduleParameters, module: siard-2 with parameters ## Parameters
**Import module:** siard-2
- file = d:\projekti\E-ARK\out_pg1.siard
2016-08-25 12:05:20,724 [main] DEBUG (c.d.m.Reporter) moduleParameters, module: postgresql with parameters **Export module:** postgresql
- hostname = 10.0.92.38
- database = eark_testB
- username = pg_eark
- password = <filtered>
- port number = 5432
2016-08-25 12:05:20,729 [main] INFO  (c.d.Main) Converting database: siard-2 to postgresql
2016-08-25 12:05:20,757 [main] DEBUG (c.d.m.j.o.JDBCExportModule) Database: postgres
2016-08-25 12:05:20,757 [main] DEBUG (c.d.m.j.o.JDBCExportModule) Loading JDBC Driver org.postgresql.Driver
2016-08-25 12:05:20,762 [main] DEBUG (c.d.m.j.o.JDBCExportModule) Getting admin connection
2016-08-25 12:05:20,817 [main] DEBUG (c.d.m.j.o.JDBCExportModule) Connected
2016-08-25 12:05:20,834 [main] INFO  (c.d.m.p.o.PostgreSQLJDBCExportModule) Target database does not exist. Creating database eark_testB
2016-08-25 12:05:20,836 [main] DEBUG (c.d.m.Reporter) customMessage, invoker: com.databasepreservation.modules.postgreSql.out.PostgreSQLJDBCExportModule; message: - Information: target database with name `eark_testB` did not exist and was created
2016-08-25 12:05:20,837 [main] DEBUG (c.d.m.j.o.JDBCExportModule) Database: postgres
2016-08-25 12:05:20,837 [main] DEBUG (c.d.m.j.o.JDBCExportModule) Loading JDBC Driver org.postgresql.Driver
2016-08-25 12:05:20,837 [main] DEBUG (c.d.m.j.o.JDBCExportModule) Getting admin connection
2016-08-25 12:05:20,846 [main] DEBUG (c.d.m.j.o.JDBCExportModule) Connected
2016-08-25 12:05:21,825 [main] DEBUG (c.d.m.Reporter) dataTypeChangedOnImport, invoker: com.databasepreservation.modules.siard.in.metadata.typeConverter.SQL99StandardDatatypeImporter; message: Type conversion in import module: in `drivers.persons.id` (format: schema.table.column) has an unidentified original type and was converted to the standard type `INTEGER`; and type: Type{description='null', originalTypeName='null', sql99TypeName='INTEGER', sql2008TypeName='INTEGER'}
2016-08-25 12:05:21,826 [main] DEBUG (c.d.m.Reporter) dataTypeChangedOnImport, invoker: com.databasepreservation.modules.siard.in.metadata.typeConverter.SQL99StandardDatatypeImporter; message: Type conversion in import module: in `drivers.persons.name` (format: schema.table.column) has an unidentified original type and was converted to the standard type `CHARACTER VARYING(2147483647)`; and type: Type{description='null', originalTypeName='null', sql99TypeName='CHARACTER VARYING(2147483647)', sql2008TypeName='CHARACTER VARYING(2147483647)'}
2016-08-25 12:05:21,826 [main] DEBUG (c.d.m.Reporter) dataTypeChangedOnImport, invoker: com.databasepreservation.modules.siard.in.metadata.typeConverter.SQL99StandardDatatypeImporter; message: Type conversion in import module: in `drivers.persons.picture` (format: schema.table.column) has an unidentified original type and was converted to the standard type `BINARY LARGE OBJECT`; and type: Type{description='null', originalTypeName='null', sql99TypeName='BINARY LARGE OBJECT', sql2008TypeName='BINARY LARGE OBJECT'}
2016-08-25 12:05:21,828 [main] DEBUG (c.d.m.j.o.JDBCExportModule) Loading JDBC Driver org.postgresql.Driver
2016-08-25 12:05:21,828 [main] DEBUG (c.d.m.j.o.JDBCExportModule) Getting connection
2016-08-25 12:05:21,839 [main] DEBUG (c.d.m.j.o.JDBCExportModule) Connected
2016-08-25 12:05:21,851 [main] INFO  (c.d.m.j.o.JDBCExportModule) Exporting database structure
2016-08-25 12:05:21,852 [main] INFO  (c.d.m.j.o.JDBCExportModule) Exporting schema structure for schema drivers
2016-08-25 12:05:21,855 [main] DEBUG (c.d.m.j.o.JDBCExportModule) Exporting schema structure for drivers
2016-08-25 12:05:21,860 [main] DEBUG (c.d.m.j.o.JDBCExportModule) batch executed: drivers
2016-08-25 12:05:21,860 [main] INFO  (c.d.m.j.o.JDBCExportModule) Exporting table structure for persons
2016-08-25 12:05:21,862 [main] DEBUG (c.d.m.j.o.JDBCExportModule) Adding to batch creation of table persons
2016-08-25 12:05:21,862 [main] DEBUG (c.d.m.p.PostgreSQLHelper) Checking PSQL type int4
2016-08-25 12:05:21,863 [main] DEBUG (c.d.m.p.PostgreSQLHelper) Checking PSQL type varchar
2016-08-25 12:05:21,863 [main] DEBUG (c.d.m.p.PostgreSQLHelper) Using PostgreSQL original type varchar
2016-08-25 12:05:21,863 [main] DEBUG (c.d.m.Reporter) dataTypeChangedOnExport, invoker: com.databasepreservation.modules.postgreSql.PostgreSQLHelper; message: - Type conversion in export module: in `drivers.persons.name` (format: schema.table.column) has original type `VARCHAR` and standard type `CHARACTER VARYING(2147483647)`, will be created as `varchar` in the target database; and column: ColumnStructure [id=drivers.persons.name, name=name, type=Type{description='null', originalTypeName='varchar', sql99TypeName='CHARACTER VARYING(2147483647)', sql2008TypeName='CHARACTER VARYING(2147483647)'}, defaultValue=null, nillable=true, description=null, isAutoIncrement=null]
2016-08-25 12:05:21,863 [main] DEBUG (c.d.m.p.PostgreSQLHelper) Checking PSQL type bytea
2016-08-25 12:05:21,863 [main] DEBUG (c.d.m.Reporter) dataTypeChangedOnExport, invoker: com.databasepreservation.modules.postgreSql.PostgreSQLHelper; message: - Type conversion in export module: in `drivers.persons.picture` (format: schema.table.column) has original type `BYTEA` and standard type `BINARY LARGE OBJECT`, will be created as `bytea` in the target database; and column: ColumnStructure [id=drivers.persons.picture, name=picture, type=Type{description='null', originalTypeName='bytea', sql99TypeName='BINARY LARGE OBJECT', sql2008TypeName='BINARY LARGE OBJECT'}, defaultValue=null, nillable=true, description=null, isAutoIncrement=null]
2016-08-25 12:05:21,863 [main] DEBUG (c.d.m.j.o.JDBCExportModule) SQL: CREATE TABLE "drivers"."persons" ("id" integer NOT NULL, "name" varchar NULL, "picture" bytea NULL)
2016-08-25 12:05:21,863 [main] DEBUG (c.d.m.p.PostgreSQLHelper) Checking PSQL type int4
2016-08-25 12:05:21,863 [main] DEBUG (c.d.m.p.PostgreSQLHelper) Checking PSQL type varchar
2016-08-25 12:05:21,864 [main] DEBUG (c.d.m.p.PostgreSQLHelper) Using PostgreSQL original type varchar
2016-08-25 12:05:21,864 [main] DEBUG (c.d.m.Reporter) dataTypeChangedOnExport, invoker: com.databasepreservation.modules.postgreSql.PostgreSQLHelper; message: - Type conversion in export module: in `drivers.persons.name` (format: schema.table.column) has original type `VARCHAR` and standard type `CHARACTER VARYING(2147483647)`, will be created as `varchar` in the target database; and column: ColumnStructure [id=drivers.persons.name, name=name, type=Type{description='null', originalTypeName='varchar', sql99TypeName='CHARACTER VARYING(2147483647)', sql2008TypeName='CHARACTER VARYING(2147483647)'}, defaultValue=null, nillable=true, description=null, isAutoIncrement=null]
2016-08-25 12:05:21,864 [main] DEBUG (c.d.m.p.PostgreSQLHelper) Checking PSQL type bytea
2016-08-25 12:05:21,864 [main] DEBUG (c.d.m.Reporter) dataTypeChangedOnExport, invoker: com.databasepreservation.modules.postgreSql.PostgreSQLHelper; message: - Type conversion in export module: in `drivers.persons.picture` (format: schema.table.column) has original type `BYTEA` and standard type `BINARY LARGE OBJECT`, will be created as `bytea` in the target database; and column: ColumnStructure [id=drivers.persons.picture, name=picture, type=Type{description='null', originalTypeName='bytea', sql99TypeName='BINARY LARGE OBJECT', sql2008TypeName='BINARY LARGE OBJECT'}, defaultValue=null, nillable=true, description=null, isAutoIncrement=null]
2016-08-25 12:05:21,864 [main] DEBUG (c.d.m.j.o.JDBCExportModule) SQL: ALTER TABLE "drivers"."persons" ADD CONSTRAINT "primary_index" PRIMARY KEY ("id")
2016-08-25 12:05:21,864 [main] DEBUG (c.d.m.j.o.JDBCExportModule) Exporting schema structure drivers completed
2016-08-25 12:05:21,864 [main] INFO  (c.d.m.j.o.JDBCExportModule) Exporting schema structure for schema drivers
2016-08-25 12:05:21,866 [main] DEBUG (c.d.m.j.o.JDBCExportModule) Executing table creation batch
2016-08-25 12:05:21,882 [main] INFO  (c.d.m.j.o.JDBCExportModule) Exporting database structure finished
2016-08-25 12:05:21,890 [main] INFO  (c.d.m.s.i.c.SIARD2ContentImportStrategy) Obtaining contents from table 'drivers.persons'
2016-08-25 12:05:21,891 [main] DEBUG (c.d.m.j.o.JDBCExportModule) Started data open: drivers.persons
2016-08-25 12:05:21,892 [main] INFO  (c.d.m.j.o.JDBCExportModule) Exporting content to table for drivers.persons
2016-08-25 12:05:21,900 [main] DEBUG (c.d.m.j.o.JDBCExportModule) sql: INSERT INTO "drivers"."persons" VALUES (?, ?, ?)
2016-08-25 12:05:21,909 [main] ERROR (c.d.Main) Fatal error: Unexpected exception (null)
java.lang.NullPointerException: null
	at com.databasepreservation.modules.siard.in.read.ZipAndFolderReadStrategy.createInputStream(ZipAndFolderReadStrategy.java:36)
	at com.databasepreservation.modules.siard.in.content.SIARD2ContentImportStrategy.startElement(SIARD2ContentImportStrategy.java:234)
	at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.startElement(Unknown Source)
	at com.sun.org.apache.xerces.internal.parsers.AbstractXMLDocumentParser.emptyElement(Unknown Source)
	at com.sun.org.apache.xerces.internal.impl.xs.XMLSchemaValidator.emptyElement(Unknown Source)
	at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.scanStartElement(Unknown Source)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl$FragmentContentDriver.next(Unknown Source)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(Unknown Source)
	at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(Unknown Source)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(Unknown Source)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(Unknown Source)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(Unknown Source)
	at com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(Unknown Source)
	at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(Unknown Source)
	at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source)
	at com.databasepreservation.modules.siard.in.content.SIARD2ContentImportStrategy.importContent(SIARD2ContentImportStrategy.java:141)
	at com.databasepreservation.modules.siard.in.input.SIARDImportDefault.getDatabase(SIARDImportDefault.java:45)
	at com.databasepreservation.Main.run(Main.java:126)
	at com.databasepreservation.Main.internal_main(Main.java:80)
	at com.databasepreservation.Main.main(Main.java:62)
2016-08-25 12:05:21,913 [main] INFO  (c.d.m.Reporter) A report was generated with a listing of information that was modified during the conversion.
2016-08-25 12:05:21,915 [main] INFO  (c.d.m.Reporter) The report file is located at d:\projekti\E-ARK\dbptk-report-20160825120520712.txt
2016-08-25 12:05:21,918 [main] INFO  (c.d.Main) Please report any problems at https://github.com/keeps/db-preservation-toolkit/issues/new
2016-08-25 12:05:21,921 [main] DEBUG (c.d.Main) #########################################################
2016-08-25 12:05:21,922 [main] DEBUG (c.d.Main) #   FINISH-ID-43439e0b-27f5-48ff-9a66-3c49438826f6
2016-08-25 12:05:21,922 [main] DEBUG (c.d.Main) #   FINISH v2.0.0-beta5
2016-08-25 12:05:21,922 [main] DEBUG (c.d.Main) #   EXIT CODE: 1
2016-08-25 12:05:21,922 [main] DEBUG (c.d.Main) #########################################################
```

**Comment by chalkos** [comment]
Could you check if this also happens when exporting a package created (from PostgreSQL) **without** option `--export-external-lobs` to postgreSQL.

I had a similar problem while I was improving the handling of LOBs and I fixed it (but I haven't released it yet). I'm trying to understand if this is a different problem or the one I already fixed.

**Comment by BorisDomajnko** [comment]
As I wrote above, without --export-external-lobs it works and all columns are re-created in the target environment.

**Comment by chalkos** [comment]
Sorry, I did not understand that at first, I thought all the steps were needed to reproduce the issue. Thank you for testing different scenarios.

Please try using this version [dbptk-app-2.0.0-SNAPSHOT-255-1.zip](https://github.com/keeps/db-preservation-toolkit/files/436860/dbptk-app-2.0.0-SNAPSHOT-255-1.zip) to see if the problem is fixed in that version.

**Comment by BorisDomajnko** [comment]
TEST1:
- Creation of a SIARD-2 package (from PostgreSQL): exception, but package created
- Restore to (to PostgreSQL): OK
TEST2:
- Creation a package (from PostgreSQL) with option: --export-external-lobs: OK
- Restore to (to PostgreSQL):  ERROR Fatal error: Unexpected exception (null)

TEST1:
d:\projekti\E-ARK>java -Xmx1000m -Dfile.encoding=UTF-8 -jar dbptk-app-2.0.0-SNAPSHOT-255-1.jar -i postgresql -ih pluton -idb testB -iu pg_boris -ip x -ide -ipn 5432 -e list-tables -ef out_pg1
_tables.txt
      Converting database: postgresql to list-tables
      Initializing target database
      Converting database structure
      Obtaining table structure for drivers.persons
      Database structure converted
      Getting contents from table 'drivers.persons'
      Total of 0 row(s) processed
      Obtained contents from table 'drivers.persons'
      Run time 0m 0s
      A report was generated with a listing of information that was modified during the conversion.
      The report file is located at d:\projekti\E-ARK\dbptk-report-20160825143004335.txt
      Please report any problems at https://github.com/keeps/db-preservation-toolkit/issues/new

d:\projekti\E-ARK>java -Xmx1000m -Dfile.encoding=UTF-8 -jar dbptk-app-2.0.0-SNAPSHOT-255-1.jar -i postgresql -ih pluton -idb testB -iu pg_boris -ip x -ide -ipn 5432 -e siard-2 -ef out_pg1.sia
rd
      Converting database: postgresql to siard-2
      Initializing target database
      Converting database structure
      Obtaining table structure for drivers.persons
      Database structure converted
      Getting contents from table 'drivers.persons'
      Total of 3 row(s) processed
      Obtained contents from table 'drivers.persons'
WARN  Schema public was not exported because it does not contain tables.
      Run time 0m 0s
      A report was generated with a listing of information that was modified during the conversion.
      The report file is located at d:\projekti\E-ARK\dbptk-report-20160825143034366.txt
      Please report any problems at https://github.com/keeps/db-preservation-toolkit/issues/new
Exception in thread "Thread-0" Exception in thread "Thread-2" java.lang.IllegalStateException: A PathInputStreamProvider was cleaned by a shutdown hook. Path: C:\Users\BDOMAJ~1\AppData\Local\Temp\dbpt
k5762429293467856515lob
        at com.databasepreservation.common.PathInputStreamProvider$1.run(PathInputStreamProvider.java:75)
Exception in thread "Thread-1" java.lang.IllegalStateException: A PathInputStreamProvider was cleaned by a shutdown hook. Path: C:\Users\BDOMAJ~1\AppData\Local\Temp\dbptk5052345716606756366lob
        at com.databasepreservation.common.PathInputStreamProvider$1.run(PathInputStreamProvider.java:75)
java.lang.IllegalStateException: A PathInputStreamProvider was cleaned by a shutdown hook. Path: C:\Users\BDOMAJ~1\AppData\Local\Temp\dbptk1884645526714057179lob
        at com.databasepreservation.common.PathInputStreamProvider$1.run(PathInputStreamProvider.java:75)

2016-08-25 14:34:54,539 [main] DEBUG (c.d.Main) #########################################################
2016-08-25 14:34:54,541 [main] DEBUG (c.d.Main) #   START-ID-4a338d2e-3dfc-41ea-a3fa-89d8d6c86bc6
2016-08-25 14:34:54,541 [main] DEBUG (c.d.Main) #   START v2.0.0-SNAPSHOT-255-1
2016-08-25 14:34:54,541 [main] DEBUG (c.d.Main) #########################################################
2016-08-25 14:34:54,541 [main] DEBUG (c.d.c.CLI) Operating system: Windows 7
2016-08-25 14:34:54,541 [main] DEBUG (c.d.c.CLI) Architecture: x86
2016-08-25 14:34:54,541 [main] DEBUG (c.d.c.CLI) Version: 6.1
2016-08-25 14:34:54,541 [main] DEBUG (c.d.c.CLI) Java vendor: Oracle Corporation
2016-08-25 14:34:54,541 [main] DEBUG (c.d.c.CLI) Java version: 1.8.0_101
2016-08-25 14:34:54,541 [main] DEBUG (c.d.c.CLI) Java class version: 52.0
2016-08-25 14:34:54,541 [main] DEBUG (c.d.c.CLI) Default Charset reported by java: UTF-8
2016-08-25 14:34:54,541 [main] DEBUG (c.d.c.CLI) Default Charset used by StreamWriter: UTF8
2016-08-25 14:34:54,542 [main] DEBUG (c.d.c.CLI) file.encoding property: UTF-8
2016-08-25 14:34:54,553 [main] DEBUG (c.d.m.Reporter) moduleParameters, module: postgresql with parameters ## Parameters
**Import module:** postgresql
- hostname = pluton
- database = testB
- username = pg_boris
- password = <filtered>
- port number = 5432
2016-08-25 14:34:54,566 [main] DEBUG (c.d.m.Reporter) moduleParameters, module: siard-2 with parameters **Export module:** siard-2
- file = d:\projekti\E-ARK\out_pg1.siard
- compress = false
- pretty xml = false
2016-08-25 14:34:54,575 [main] INFO  (c.d.Main) Converting database: postgresql to siard-2
2016-08-25 14:34:54,577 [main] INFO  (c.d.m.j.i.JDBCImportModule) Initializing target database
2016-08-25 14:34:54,588 [main] INFO  (c.d.m.j.i.JDBCImportModule) Converting database structure
2016-08-25 14:34:54,630 [main] DEBUG (c.d.m.p.i.PostgreSQLJDBCImportModule) Loading JDBC Driver org.postgresql.Driver
2016-08-25 14:34:54,634 [main] DEBUG (c.d.m.p.i.PostgreSQLJDBCImportModule) Getting connection
2016-08-25 14:34:54,689 [main] DEBUG (c.d.m.p.i.PostgreSQLJDBCImportModule) Connected
2016-08-25 14:34:54,696 [main] DEBUG (c.d.m.j.i.JDBCImportModule) driver version: PostgreSQL 9.4.1208.jre7
2016-08-25 14:34:54,722 [main] INFO  (c.d.m.j.i.JDBCImportModule) Obtaining table structure for drivers.persons
2016-08-25 14:34:54,733 [main] DEBUG (c.d.m.Reporter) dataTypeChangedOnImport, invoker: com.databasepreservation.modules.postgreSql.in.PostgreSQLJDBCDatatypeImporter; message: - Type conversion in import module: in `drivers.persons.id` (format: schema.table.column) has original type `INT4` and was converted to the standard type `INTEGER`; and type: Type{description='null', originalTypeName='int4', sql99TypeName='INTEGER', sql2008TypeName='INTEGER'}
2016-08-25 14:34:54,734 [main] DEBUG (c.d.m.j.i.JDBCImportModule) Data type: 4
Type name: int4
Column size: 10
Decimal digits: 0
Radix: 10
Remarks: null
Default value: null
Index: 1
Is Nullable: false
Is auto increment: false
Calculated type: SimpleTypeNumericExact
ColumnType hash: -1058864868

2016-08-25 14:34:54,736 [main] DEBUG (c.d.m.Reporter) dataTypeChangedOnImport, invoker: com.databasepreservation.modules.postgreSql.in.PostgreSQLJDBCDatatypeImporter; message: - Type conversion in import module: in `drivers.persons.name` (format: schema.table.column) has original type `VARCHAR` and was converted to the standard type `CHARACTER VARYING(2147483647)`; and type: Type{description='null', originalTypeName='varchar', sql99TypeName='CHARACTER VARYING(2147483647)', sql2008TypeName='CHARACTER VARYING(2147483647)'}
2016-08-25 14:34:54,736 [main] DEBUG (c.d.m.j.i.JDBCImportModule) Data type: 12
Type name: varchar
Column size: 2147483647
Decimal digits: 0
Radix: 10
Remarks: null
Default value: null
Index: 2
Is Nullable: true
Is auto increment: false
Calculated type: SimpleTypeString
ColumnType hash: -547542414

2016-08-25 14:34:54,736 [main] DEBUG (c.d.m.Reporter) dataTypeChangedOnImport, invoker: com.databasepreservation.modules.postgreSql.in.PostgreSQLJDBCDatatypeImporter; message: - Type conversion in import module: in `drivers.persons.picture` (format: schema.table.column) has original type `BYTEA` and was converted to the standard type `BINARY LARGE OBJECT`; and type: Type{description='null', originalTypeName='bytea', sql99TypeName='BINARY LARGE OBJECT', sql2008TypeName='BINARY LARGE OBJECT'}
2016-08-25 14:34:54,736 [main] DEBUG (c.d.m.j.i.JDBCImportModule) Data type: -2
Type name: bytea
Column size: 2147483647
Decimal digits: 0
Radix: 10
Remarks: null
Default value: null
Index: 3
Is Nullable: true
Is auto increment: false
Calculated type: SimpleTypeBinary
ColumnType hash: 1224905463

2016-08-25 14:34:54,778 [main] DEBUG (c.d.m.j.i.JDBCImportModule) count query: SELECT COUNT(*) FROM "drivers"."persons"
2016-08-25 14:34:54,779 [main] DEBUG (c.d.m.j.i.JDBCImportModule) Counted 3 rows
2016-08-25 14:34:54,794 [main] DEBUG (c.d.m.j.i.JDBCImportModule) Database structure obtained
2016-08-25 14:34:54,794 [main] INFO  (c.d.m.j.i.JDBCImportModule) Database structure converted
2016-08-25 14:34:54,796 [main] INFO  (c.d.m.j.i.JDBCImportModule) Getting contents from table 'drivers.persons'
2016-08-25 14:34:54,810 [main] DEBUG (c.d.m.p.i.PostgreSQLJDBCImportModule) query: SELECT id, name, picture FROM drivers.persons
2016-08-25 14:34:54,838 [main] INFO  (c.d.m.j.i.JDBCImportModule) Total of 3 row(s) processed
2016-08-25 14:34:54,842 [main] INFO  (c.d.m.j.i.JDBCImportModule) Obtained contents from table 'drivers.persons'
2016-08-25 14:34:54,845 [main] DEBUG (c.d.m.j.i.JDBCImportModule) Freeing resources
2016-08-25 14:34:55,101 [main] WARN  (c.d.m.s.o.m.SIARD2MetadataExportStrategy) Schema public was not exported because it does not contain tables.
2016-08-25 14:34:55,121 [main] DEBUG (c.d.m.j.i.JDBCImportModule) Closing connection to source database
2016-08-25 14:34:55,122 [main] INFO  (c.d.Main) Run time 0m 0s
2016-08-25 14:34:55,124 [main] INFO  (c.d.m.Reporter) A report was generated with a listing of information that was modified during the conversion.
2016-08-25 14:34:55,127 [main] INFO  (c.d.m.Reporter) The report file is located at d:\projekti\E-ARK\dbptk-report-20160825143454549.txt
2016-08-25 14:34:55,130 [main] INFO  (c.d.Main) Please report any problems at https://github.com/keeps/db-preservation-toolkit/issues/new
2016-08-25 14:34:55,132 [main] DEBUG (c.d.Main) #########################################################
2016-08-25 14:34:55,132 [main] DEBUG (c.d.Main) #   FINISH-ID-4a338d2e-3dfc-41ea-a3fa-89d8d6c86bc6
2016-08-25 14:34:55,132 [main] DEBUG (c.d.Main) #   FINISH v2.0.0-SNAPSHOT-255-1
2016-08-25 14:34:55,132 [main] DEBUG (c.d.Main) #   EXIT CODE: 0
2016-08-25 14:34:55,132 [main] DEBUG (c.d.Main) #########################################################

TEST2:
[dbptk-app.log.txt](https://github.com/keeps/db-preservation-toolkit/files/436994/dbptk-app.log.txt)

**Comment by chalkos** [comment]
Can you try TEST2 with this version [dbptk-app-2.0.0-SNAPSHOT-255-2.zip](https://github.com/keeps/db-preservation-toolkit/files/437099/dbptk-app-2.0.0-SNAPSHOT-255-2.zip) and send me the log? This version only adds more information to the log. the error will still happen.

**Comment by BorisDomajnko** [comment]
d:\projekti\E-ARK>java -Xmx1000m -Dfile.encoding=UTF-8  -jar  dbptk-app-2.0.0-SNAPSHOT-255-2.jar -e postgresql -eh 10.0.92.38 -edb eark_testB -eu pg_eark -ep x -ede -epn 5432 -i siard-2 -if out_
pg1.siard
      Converting database: siard-2 to postgresql
      Target database does not exist. Creating database eark_testB
      Exporting database structure
      Exporting schema structure for schema drivers
      Exporting table structure for persons
      Exporting schema structure for schema drivers
      Exporting database structure finished
      Obtaining contents from table 'drivers.persons'
      Exporting content to table for drivers.persons
SLF4J: Failed toString() invocation on an object of type [com.databasepreservation.modules.siard.common.SIARDArchiveContainer]
java.lang.NullPointerException
        at com.databasepreservation.modules.siard.common.SIARDArchiveContainer.toString(SIARDArchiveContainer.java:27)
        at org.slf4j.helpers.MessageFormatter.safeObjectAppend(MessageFormatter.java:305)
        at org.slf4j.helpers.MessageFormatter.deeplyAppendParameter(MessageFormatter.java:277)
        at org.slf4j.helpers.MessageFormatter.arrayFormat(MessageFormatter.java:231)
        at ch.qos.logback.classic.spi.LoggingEvent.getFormattedMessage(LoggingEvent.java:293)
        at ch.qos.logback.classic.spi.LoggingEvent.prepareForDeferredProcessing(LoggingEvent.java:206)
        at ch.qos.logback.core.OutputStreamAppender.subAppend(OutputStreamAppender.java:205)
        at ch.qos.logback.core.rolling.RollingFileAppender.subAppend(RollingFileAppender.java:228)
        at ch.qos.logback.core.OutputStreamAppender.append(OutputStreamAppender.java:100)
        at ch.qos.logback.core.UnsynchronizedAppenderBase.doAppend(UnsynchronizedAppenderBase.java:84)
        at ch.qos.logback.core.spi.AppenderAttachableImpl.appendLoopOnAppenders(AppenderAttachableImpl.java:48)
        at ch.qos.logback.classic.Logger.appendLoopOnAppenders(Logger.java:270)
        at ch.qos.logback.classic.Logger.callAppenders(Logger.java:257)
        at ch.qos.logback.classic.Logger.buildLoggingEventAndAppend(Logger.java:421)
        at ch.qos.logback.classic.Logger.filterAndLog_2(Logger.java:414)
        at ch.qos.logback.classic.Logger.debug(Logger.java:490)
        at com.databasepreservation.modules.siard.in.content.SIARD2ContentImportStrategy.startElement(SIARD2ContentImportStrategy.java:238)
        at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.startElement(Unknown Source)
        at com.sun.org.apache.xerces.internal.parsers.AbstractXMLDocumentParser.emptyElement(Unknown Source)
        at com.sun.org.apache.xerces.internal.impl.xs.XMLSchemaValidator.emptyElement(Unknown Source)
        at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.scanStartElement(Unknown Source)
        at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl$FragmentContentDriver.next(Unknown Source)
        at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(Unknown Source)
        at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(Unknown Source)
        at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(Unknown Source)
        at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(Unknown Source)
        at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(Unknown Source)
        at com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(Unknown Source)
        at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(Unknown Source)
        at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source)
        at com.databasepreservation.modules.siard.in.content.SIARD2ContentImportStrategy.importContent(SIARD2ContentImportStrategy.java:142)
        at com.databasepreservation.modules.siard.in.input.SIARDImportDefault.getDatabase(SIARDImportDefault.java:45)
        at com.databasepreservation.Main.run(Main.java:127)
        at com.databasepreservation.Main.internal_main(Main.java:80)
        at com.databasepreservation.Main.main(Main.java:62)
ERROR Fatal error: Unexpected exception (null)
      A report was generated with a listing of information that was modified during the conversion.
      The report file is located at d:\projekti\E-ARK\dbptk-report-20160825152437736.txt
      Please report any problems at https://github.com/keeps/db-preservation-toolkit/issues/new

[dbptk-app.log.txt](https://github.com/keeps/db-preservation-toolkit/files/437112/dbptk-app.log.txt)

**Comment by chalkos** [comment]
This seems to be a windows issue. Please try this version [dbptk-app-2.0.0-SNAPSHOT-255-3.zip](https://github.com/keeps/db-preservation-toolkit/files/437205/dbptk-app-2.0.0-SNAPSHOT-255-3.zip)

**Comment by BorisDomajnko** [comment]
Now it works!
If I used  --export-external-lobs, are the LOBS always populated into target DB, but in addition the access tool can use out_pg1.siard_lobseg_1\  files?

**Comment by chalkos** [comment]
Yes, that folder is where the large files from the database are stored as files. They are not in the out_pg1.siard file anymore.

If you had a lot of LOBs (or used different parameters), multiple folders like:

* out_pg1.siard_lobseg_1\
* out_pg1.siard_lobseg_2\
* out_pg1.siard_lobseg_3\
* ...

would be created.

I will integrate the fix in the main code and then close this issue.

**Comment by BorisDomajnko** [comment]
I suggest we modify the help text with the reminder:
now:
 -eel, --export-external-lobs    (optional) Saves any LOBs outside the siard file.
new:
 -eel, --export-external-lobs    (optional) Saves any LOBs outside the siard file. You will need to archive them separately.

**Comment by chalkos** [comment]
DBPTK converts databases to SIARD and SIARD back to DBMS. It should have no say in how SIARD (or anything else for that matter) is archived. Therefore I don't think your addition is adequate, since it assumes the archive will do or will want to do things in a certain way.

**Comment by BorisDomajnko** [comment]
You are right: created siard file and the lob folders are in the same place. From here the SIP (one or more) can be created.
In the test environment this folder is less transparent because of many repeations of test scenarios.

**Comment by BorisDomajnko** [comment]
BTW, this topic is described in "E-ARK SIP Pilot Specification" (EARK_D3_3_01.pdf), chapter 4.3 Relational databases.

**Comment by chalkos** [comment]
This has been fixed in #245
